src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -266,6 +302,19 @@ public bool TryGetSemanticModel([NotNullWhen(returnValue: true)] out SemanticMod
         /// return the same value if called multiple times.
         /// </returns>
         public async Task<SemanticModel?> GetSemanticModelAsync(CancellationToken cancellationToken = default)
+        {
+            return await GetSemanticModelHelperAsync(disableNullableAnalysis: false, cancellationToken: cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Gets the semantic model for this document asynchronously.
+        /// </summary>
+        /// <returns>
+        /// The returned <see cref="SemanticModel"/> may be <see langword="null"/> if <see
+        /// cref="SupportsSemanticModel"/> returns <see langword="false"/>. This function will
+        /// return the same value if called multiple times.
+        /// </returns>
+        private async Task<SemanticModel?> GetSemanticModelHelperAsync(bool disableNullableAnalysis = false, CancellationToken cancellationToken = default)
```

Comment:
don't use optional parameters on non-public methods.  all callers should be required to pass all this in.